### PR TITLE
Leverage SDK v2 context-aware schema functions and diagnostics

### LINKED ISSUE #175: Leverage SDK v2 Context-Aware Schema Functions and Diagnostics
Once upgrading is complete in https://github.com/grafana/terraform-provider-grafana/issues/159, these will be easy to implement across all resources:

- [x] [context-aware schema functions](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#more-support-for-context-context)
- [x] [diagnostics](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#support-for-diagnostics)

### PR DESCRIPTION
**Pull request by nikosmeds**
Leverage support for [context-aware schema functions](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#more-support-for-context-context) and [diagnostics](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#support-for-diagnostics), which were made available in Terraform Plugin SDK v2.

Closes https://github.com/grafana/terraform-provider-grafana/issues/175.

### CODE REVIEW

File: grafana/resource_organization.go [hunk 1]
```diff
@@ -151,15 +165,15 @@ func ExistsOrganization(d *schema.ResourceData, meta interface{}) (bool, error)
...
 	if err != nil || !exists {
 		return nil, errors.New(fmt.Sprintf("Error: Unable to import Grafana Organization: %s.", err))
 	}
 	d.Set("admin_user", "admin")
 	d.Set("create_users", "true")
-	err = ReadOrganization(d, meta)
-	if err != nil {
+	diags := ReadOrganization(ctx, d, meta)
+	if diags != nil {
 		return nil, err
```
Comment (julienduchesne) [inline grafana/resource_organization.go]: Unless I'm misunderstanding, I think you'll be returning the wrong `err` here

File: grafana/resource_team.go [hunk 2]
```diff
@@ -129,13 +143,13 @@ func ExistsTeam(d *schema.ResourceData, meta interface{}) (bool, error) {
...
+func ImportTeam(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
 	exists, err := ExistsTeam(d, meta)
 	if err != nil || !exists {
 		return nil, errors.New(fmt.Sprintf("Error: Unable to import Grafana Team: %s.", err))
 	}
-	err = ReadTeam(d, meta)
-	if err != nil {
+	diags := ReadTeam(ctx, d, meta)
+	if diags != nil {
 		return nil, err
```
Comment (julienduchesne) [inline grafana/resource_team.go]: Same thing here as above

File: grafana/resource_user.go [hunk 3]
```diff
@@ -135,13 +141,13 @@ func ExistsUser(d *schema.ResourceData, meta interface{}) (bool, error) {
...
+func ImportUser(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
 	exists, err := ExistsUser(d, meta)
 	if err != nil || !exists {
 		return nil, errors.New(fmt.Sprintf("Error: Unable to import Grafana User: %s.", err))
 	}
-	err = ReadUser(d, meta)
-	if err != nil {
+	diags := ReadUser(ctx, d, meta)
+	if diags != nil {
 		return nil, err
```
Comment (julienduchesne) [inline grafana/resource_user.go]: Here too

File: grafana/resource_organization.go (hunk 1 above)
Comment (julienduchesne) [reply grafana/resource_organization.go]: As discussed with @nikosmeds, this is probably just a case where we shouldn't call Read in the Import function. The importer function should only do the work necessary to set the ID and then the Read will be called implicitly when the state is refreshed following the import. With the read call going away, this function won't have to deal with `Diagnostics`

**Review by inkel — APPROVED** [review]

**Review by julienduchesne — APPROVED** [review]

**Review by trotttrotttrott — APPROVED** [review]
Thanks @nikosmeds! 👏

**Pull request merged**